tests/50federation/40devicelists.pl

```diff
@@ -608,6 +628,19 @@
                Future->done(1)
             })
          );
+      })->then( sub {
+         # Synapse with workers would often fail this test, because
+         # - the device list info retrieved over federation was written to DB
+         #   by the master worker
+         # - the master worker notifies the others over replication, so the
+         #   workers can invalidate their caches
+         # - the second request (that we're about to send out) would sometimes
+         #   arrive at the stream_writer worker before it'd been notified over
+         #   replication
+         #
+         # Sync so that we have a chance for the replication to propagate.
+         sync_until_user_in_device_list($local_user, new_User(user_id=>$outbound_client_user, http=>"", server_name=>""));
```

Comment:
```suggestion
         sync_until_user_in_device_list( $local_user,
            new_User( user_id => $outbound_client_user, http => "", server_name => "" )
         );
```

I wonder if instead we should change `sync_until_user_in_device_list` to accept a user ID as the second param? Probably a lot nicer?

Reply:
I'd be tempted to make it accept either a `User` struct or a user ID, if only because I don't want to touch all of the other call sites. I'm not sure if there's a good perl idiom for that?